### PR TITLE
Fixing the instantiation of customized builtin elements (related to custom elements)

### PR DESCRIPTION
**Pull request by joeldenning**
The [custom element spec](https://html.spec.whatwg.org/multipage/scripting.html#custom-elements), combined with the [document.createElement](https://dom.spec.whatwg.org/#dom-document-createelement) spec explain that in order to create a DOM element as a [customized built-in element](https://html.spec.whatwg.org/multipage/scripting.html#customized-built-in-element), you should call document.createElement with an object as the second parameter. Customized builtin elements are a type of custom element that allow you to extend native DOM elements. However, React currently provides the second parameter to `document.createElement` as a string instead of an object. As far as I can tell, that is just a bug.

I am not sure of the full history of the custom elements spec, and so it might be possible that the spec used to say the second argument to createElement should be a string. I'm also not sure if anybody is depending on the behavior being like it is right now, or if this change would be considered a breaking change or not for React. My thoughts are that it is a bug fix, but I may be lacking some context here.

### CODE REVIEW

File: src/renderers/dom/stack/client/ReactDOMComponent.js [hunk 1]
```diff
@@ -552,7 +552,7 @@ ReactDOMComponent.Mixin = {
           div.innerHTML = `<${type}></${type}>`;
           el = div.removeChild(div.firstChild);
         } else if (props.is) {
-          el = ownerDocument.createElement(type, props.is);
+          el = ownerDocument.createElement(type, {is: props.is});
```
Comment (joeldenning) [inline src/renderers/dom/stack/client/ReactDOMComponent.js]: See https://dom.spec.whatwg.org/#dom-document-createelement.

The `is` prop indicates that the react element should be a customized builtin element.

File: src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js [hunk 2]
```diff
@@ -1045,7 +1045,7 @@ describe('ReactDOMComponent', () => {
         ReactDOM.render(<div is="custom-div" />, container);
         expect(document.createElement).toHaveBeenCalledWith(
           'div',
-          'custom-div',
+          {is: 'custom-div'},
```
Comment (joeldenning) [inline src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js]: `document.createElement` should not have a second argument of a string, but rather an `options` object

**Comment by joeldenning** [comment]
Updated this to fix the build -- I had forgotten to `npm run prettier` before creating this pr.

**Comment by joeldenning** [comment]
@gaearon thanks for the help -- I have updated ReactDOMFiberComponent as suggested, but the fiber tests are failing for me locally. The same tests fail for me whether I'm on the master branch or on my branch, so I don't think it's actually related to my code changes. So what I have done is push my change to ReactDOMFiberComponent without also adding my changes to `scripts/fiber/tests-failing.txt` and `scripts/fiber/tests-passing.txt`. I am hoping that maybe the CI environment will be better configured or something so that the tests pass there even though they failed for me locally.

**Comment by gaearon** [comment]
>I am not sure of the full history of the custom elements spec, and so it might be possible that the spec used to say the second argument to createElement should be a string

Pretty sure this is the case. [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement):

>For backwards compatibility with previous versions of the Custom Elements specification, some browsers will allow you to pass a string here instead of an object, where the string's value is the custom element's tag name.

The feature itself was originally added in https://github.com/facebook/react/pull/6570, and I don’t think it was extensively tested or updated to match the latest spec.

For safety I’ll merge this to master but won’t backport to 15.x. Thanks!

**Pull request merged**